tests/unit_tests/test_prompt.py

```diff
@@ -139,3 +159,18 @@ def test_substitute_constants(prompt_str, final_prompt):
     """Test substituting constants in a prompt."""
     prompt = gd.Prompt(prompt_str)
     assert prompt.source == final_prompt
+
+
+@pytest.mark.parametrize(
+    "text, expected_result",
+    [
+        (RAIL_WITH_OLD_CONSTANT_SCHEMA, True),  # Test with a single match
+        (
+            RAIL_WITH_FORMAT_INSTRUCTIONS,
+            False,
+        ),  # Test with no matches/correct namespacing
+    ],
+)
+def test_uses_old_constant_schema(text, expected_result):
+    guard = gd.Guard.from_rail_string(text)
+    assert guard.prompt.uses_old_constant_schema(text) == expected_result
```

Comment:
Can we add a spy on `warnings.warn` and assert it's being called with the expected message?